src/main/java/com/oracle/nosql/spring/data/core/mapping/NosqlKey.java

```diff
@@ -0,0 +1,46 @@
+/*-
+ * Copyright (c) 2020, 2023 Oracle and/or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Universal Permissive License v 1.0 as shown at
+ *  https://oss.oracle.com/licenses/upl/
+ */
+
+package com.oracle.nosql.spring.data.core.mapping;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import static com.oracle.nosql.spring.data.Constants.NOTSET_PRIMARY_KEY_ORDER;
+import static com.oracle.nosql.spring.data.Constants.NOTSET_SHARD_KEY;
+
+/**
+ * Identifies the annotated field as a primary key of the composite
+ * primary key.
+ */
```

Comment:
Add a @ since 1.6 javadoc annotation.

Reply:
Done

Reply:
User should be knowing the concept of shard and non shard key right? Why we need to explain the concept here

Reply:
No need to explain the concept here, it just needs to point that order matters so the user needs to know what order to choose.

Reply:
thinking of adding 'Since order of the primary key is crucial for distribution of data It is recommended to use both shard and order elements when providing this annotation'
Please suggest